app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -3,27 +3,17 @@ class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
 
   def facebook
     @user = User.from_omniauth(request.env["omniauth.auth"])
-
-    if @user.persisted?
-      sign_in_and_redirect @user, :event => :authentication
-      set_flash_message(:notice, :success, :kind => "Facebook") if is_navigational_format?
-    else
-      session["devise.facebook_data"] = request.env["omniauth.auth"]
-      redirect_to new_user_registration_url
-    end
+    
+    sign_in_and_redirect @user, :event => :authentication
+    set_flash_message(:notice, :success, :kind => "Facebook") if is_navigational_format?
   end
 
   def saml
     @user = User.from_omniauth(request.env["omniauth.auth"])
 
-    if @user.persisted?
-      c = Rails.application.config.x.firestarter_settings
-      sign_in_and_redirect @user, event: :authentication #this will throw if @user is not activated
-      set_flash_message(:notice, :success, kind: c['saml_human_name']) if is_navigational_format?
-    else
-      session["devise.saml_data"] = request.env["omniauth.auth"]
-      redirect_to new_user_registration_url
-    end
+    c = Rails.application.config.x.firestarter_settings
```

Comment:
We have a helper for this now; should be `app_settings('saml_human_name')`.
```
set_flash_message :notice, :success, kind: app_settings('saml_human_name')
```

You can include the `AppSettings` concern in this controller if that doesn't work straight out of the box.